### PR TITLE
Harden test coverage for programs with mixed course modes.

### PR DESCRIPTION
**Pull request by jimabramson**
ECOM-3635

@rlucioni @schenedx please review.

### CODE REVIEW

**Comment by schenedx** [comment]
LGTM

File: programs/apps/programs/tests/test_utils.py [hunk 1]
```diff
@@ -90,3 +92,96 @@ def setUp(self):
...
+    def test_different_modes_in_different_programs(self):
+        """
+        Explicitly validates that program completion returns a correct result
+        when two programs contain different modes of the same course run.
+        """
+        # Set all of the second program's course runs to match those of the
+        # first, and change the mode of one of those runs from 'verified' to
+        # 'honor' in the second program.
+        for run_mode in ProgramCourseRunMode.objects.filter(program_course_code__program__id=2):
+            run_mode.course_key = re.sub('org_1', 'org_0', run_mode.course_key)
```
Comment (rlucioni) [inline programs/apps/programs/tests/test_utils.py]: You can use Python's string `replace` method to do this without `re`.
```python
run_mode.course_key = run_mode.course_key.replace('org_1', 'org_0')
```

**Comment by rlucioni** [comment]
One nit aside, :+1: .

**Comment by jimabramson** [comment]
@rlucioni thanks - addressed your feedback.

**Pull request merged**